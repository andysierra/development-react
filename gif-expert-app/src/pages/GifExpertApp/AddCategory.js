import {useState} from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({setCategories}) => { 

  const [inputValue, setInputValue] = useState('');

  const onChange = ({target}) => setInputValue(target.value);

  const handleSubmit = e => {
    e.preventDefault();
    if(inputValue.trim().length > 2) {
      setCategories(cats => cats.includes(inputValue)? cats : [...cats, inputValue]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type        = "text"
        className   = "form-control"
        value       = {inputValue}
        pattern     = "^.{3,}$"
        required    = {true}
        title       = "Debe tener mÃ­nimo 3 letras"
        placeholder = 'Buscar un GIF'
        onChange    = {onChange}/>
    </form>
  );

};

AddCategory.propTypes = {
  setCategories : PropTypes.func.isRequired
}

export default AddCategory;