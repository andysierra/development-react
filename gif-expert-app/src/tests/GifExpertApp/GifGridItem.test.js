import {shallow} from 'enzyme';
import { GifGridItem } from '../../pages/GifExpertApp/GifGridItem';

describe('Tests over GifGridItem', ()=>{

  // Arrange
  const id = 'some id';
  const title = 'some title';
  const url = 'https://media2.giphy.com/media/zj6NmJC6trBhm/giphy.gif?'+
  'cid=c0222e32lx6l5f4wy8buqpwvenpm9bi8du5loy8s3klv7byz&rid=giphy.gif&ct=g';
  
  const onload = ()=>{console.log('loaded!')};
  const reactTag 
  = <GifGridItem 
    id      = {id} 
    title   = {title} 
    url     = {url} 
    onload  = {onload}
  />
  const wrapper = shallow (reactTag);

  test('should render the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
    const img = wrapper.find('img');
    expect(img.prop('alt')).toBe(id+title);
    expect(img.prop('src')).toBe(url);
    expect(wrapper.find('p').text()).toBe(title);
  });

  test('it is a card that have card-body and card-footer', () => {
    expect(wrapper.find('.card').length).toBeGreaterThan(0);
    expect(wrapper.find('.card-body').length).toBeGreaterThan(0);
    expect(wrapper.find('.card-footer').length).toBeGreaterThan(0);
  })
});