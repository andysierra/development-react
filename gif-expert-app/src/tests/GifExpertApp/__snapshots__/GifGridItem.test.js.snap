// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests over GifGridItem should render the component correctly 1`] = `
<div
  className="card p-0 col-12 col-sm-6 col-md-3 col-lg-3 col-xl-3"
>
  <div
    className="card-body p-0 d-flex flex-col justify-content-center overflow-hidden"
  >
    <img
      alt="some idsome title"
      height="200px"
      onLoad={[Function]}
      src="https://media2.giphy.com/media/zj6NmJC6trBhm/giphy.gif?cid=c0222e32lx6l5f4wy8buqpwvenpm9bi8du5loy8s3klv7byz&rid=giphy.gif&ct=g"
    />
  </div>
  <div
    className="card-footer"
  >
    <p>
      some title
    </p>
  </div>
</div>
`;
